packages/@aws-cdk/aws-msk/lib/cluster.ts

```diff
@@ -613,9 +620,20 @@ export class Cluster extends ClusterBase {
 
     let clientAuthentication;
     if (props.clientAuthentication?.saslProps?.iam) {
-      clientAuthentication = {
-        sasl: { iam: { enabled: props.clientAuthentication.saslProps.iam } },
-      };
+      if (props.clientAuthentication?.tlsProps) {
+        clientAuthentication = {
+          sasl: { iam: { enabled: props.clientAuthentication.saslProps.iam } },
+          tls: {
+            certificateAuthorityArnList: props.clientAuthentication?.tlsProps?.certificateAuthorities?.map(
+              (ca) => ca.certificateAuthorityArn,
+            ),
+          },
+        };
+      } else {
+        clientAuthentication = {
+          sasl: { iam: { enabled: props.clientAuthentication.saslProps.iam } },
+        };
```

Comment:
[nit] `sasl` always gets set, it does not have to be in an `if-else`. I'd say pull it put and only `if` the `tls` properties.